storage/dvo_recommendations_storage.go

```diff
@@ -371,27 +370,41 @@ func (storage DVORecommendationsDBStorage) updateReport(
 }
 
 func mapWorkloadRecommendations(recommendations *[]types.WorkloadRecommendation) (
-	map[string]string, map[string]int, int) {
+	map[string]string, map[string]int, map[string]int,
+) {
 	// map the namespace ID to the namespace name
 	namespaceMap := make(map[string]string)
 	// map the number of different workloads in the report per namespace
 	objectsMap := make(map[string]int)
-	nRecommendations := len(*recommendations)
+	// how many recommendations hit per namespace
+	namespaceRecommendationCount := make(map[string]int)
 
 	for _, recommendation := range *recommendations {
+		// objectsMapPerRecommendation is used to calculate number of rule hits in namespace
+		objectsMapPerRecommendation := make(map[string]int)
+
 		for _, workload := range recommendation.Workloads {
 			if _, ok := namespaceMap[workload.NamespaceUID]; !ok {
 				// store the namespace name in the namespaceMap if it's not already there
 				namespaceMap[workload.NamespaceUID] = workload.Namespace
 			}
+
 			if _, ok := objectsMap[workload.NamespaceUID]; !ok {
```

Comment:
we can also use `objectsMap[workload.NamespaceUID]++` instead of this if else here, right?